soh/soh/Enhancements/Fixes/DekuNutUpgradeFix.cpp

```diff
@@ -0,0 +1,36 @@
+#include <libultraship/bridge.h>
+#include "soh/Enhancements/game-interactor/GameInteractor_Hooks.h"
+#include "soh/ShipInit.hpp"
+#include "functions.h"
+#include "macros.h"
+#include "variables.h"
+#include "z64save.h"
+
+static constexpr int32_t CVAR_NUT_UPGRADE_FIX_DEFAULT = 0;
+#define CVAR_NUT_UPGRADE_FIX_NAME CVAR_ENHANCEMENT("DekuNutUpgradeFix")
+#define CVAR_NUT_UPGRADE_FIX_VALUE CVarGetInteger(CVAR_NUT_UPGRADE_FIX_NAME, CVAR_NUT_UPGRADE_FIX_DEFAULT)
+
+void DekuNutUpgradeFixAtForestStage(bool* should) {
+    s32 expectedNutUpgrades = (INV_CONTENT(ITEM_NUT) == ITEM_NUT ? 1 : 0) +
+                              (Flags_GetInfTable(INFTABLE_BOUGHT_NUT_UPGRADE) ? 1 : 0) +
+                              (Flags_GetItemGetInf(ITEMGETINF_OBTAINED_NUT_UPGRADE_FROM_STAGE) ? 1 : 0);
+    s32 actualNutUpgrades = CUR_UPG_VALUE(UPG_NUTS);
+
+    if (expectedNutUpgrades != actualNutUpgrades) {
```

Comment:
This hook is also missing a check for the `Player_GetMask(play) != PLAYER_MASK_SKULL`

This is necessary due to intentional `fallthrough` on the `switch (Player_GetMask(play))` in the vanilla src

Reply:
Ah. I was wondering why that extra check was there. I didn't see the intentional fallthrough.